sapi/fuzzing/config.m4

```diff
@@ -0,0 +1,61 @@
+AC_MSG_CHECKING(for CLANG fuzzer SAPI)
+
+PHP_ARG_ENABLE(fuzzer,,
+[  --enable-fuzzer          Build PHP as clang fuzzing test module (for developers], no)
+
+dnl For newer clang versions see https://llvm.org/docs/LibFuzzer.html#fuzzer-usage
+dnl for relevant flags.
+
+dnl Macro to define fuzzing target
+dnl PHP_FUZZER_TARGET(name, target-var)
+dnl
+AC_DEFUN([PHP_FUZZER_TARGET], [
+  PHP_FUZZER_BINARIES="$PHP_FUZZER_BINARIES $SAPI_FUZZER_PATH/php-fuzz-$1"
+  PHP_SUBST($2)
+  PHP_ADD_SOURCES_X([sapi/fuzzer],[fuzzer-$1.c fuzzer-sapi.c],[],$2)
+])
+
+if test "$PHP_FUZZER" != "no"; then
+  PHP_REQUIRE_CXX()
+  PHP_ADD_MAKEFILE_FRAGMENT($abs_srcdir/sapi/fuzzer/Makefile.frag)
+  SAPI_FUZZER_PATH=sapi/fuzzer
+  PHP_SUBST(SAPI_FUZZER_PATH)
+  if test -z "$LIB_FUZZING_ENGINE"; then
+    FUZZING_LIB="-lFuzzer"
+    FUZZING_CC="$CC"
+    AX_CHECK_COMPILE_FLAG([-fsanitize=address], [
+      CFLAGS="$CFLAGS -fsanitize=address"
+      CXXFLAGS="$CXXFLAGS -fsanitize=address"
+      LDFLAGS="$LDFLAGS -fsanitize=address"
```

Comment:
It seems odd that this is hardcoded to `-fsanitize=address`. What about `-fsanitize=undefined`?

Reply:
We could add that option too. On Oss-fuzz, the build auto-adds these, but on standalone, this wasn't implemented yet. Certainly can be done, but I would suggest getting the MVP version merged first and then iterate on it.

Reply:
Sounds reasonable. We may want to make this a separate option (`--enable-sanitizers=address,undefined` or so), as we also use sanitizers for non-fuzzing builds in CI.